hudi-client/hudi-client-common/src/test/java/org/apache/hudi/table/upgrade/TestEightToSevenDowngradeHandler.java

```diff
@@ -173,4 +181,38 @@ void testPropertyDowngrade() {
     assertFalse(tablePropsToAdd.containsKey(KEY_GENERATOR_TYPE));
     assertFalse(tablePropsToAdd.containsKey(KEY_GENERATOR_CLASS_NAME));
   }
+
+  @ParameterizedTest
+  @CsvSource({
+      "com.example.CustomPayload, CUSTOM, com.example.CustomPayload",
+      ", CUSTOM, com.example.CustomPayload",
```

Comment:
Should this throw and not specify the expected payload class to avoid confusion?

Reply:
ack. 
lets also remove strategy Id during downgrade